Kerberos.NET/Cache/FileHandle.cs

```diff
@@ -41,7 +45,21 @@ public FileHandle(string file, FileMode mode, FileAccess access, FileShare share
 
         public FileStream OpenStream()
         {
-            return File.Open(this.file, this.mode, this.access, this.share);
+            var fs = File.Open(this.file, this.mode, this.access, this.share);
+            // When we create file on Unix we should make sure only current user has access to it.
+            if (this.access == FileAccess.Write && SetUnixFileMode != null)
+            {
+                const int UserRead = 256;
+                const int UserWrite = 128;
+
+                try
+                {
+                    SetUnixFileMode.Invoke(null, new object[] { fs.SafeFileHandle, (int)(UserRead | UserWrite) });
```

Comment:
Probably switch to a null check `SetUnixFileMode?.Invoke` so it doesn't polute first chance exceptions noise.

Reply:
that should be guarded by the `SetUnixFileMode != null` above, right? (unless I missed your point)